### PR TITLE
Add Gradle build tool

### PR DESCRIPTION
**Pull request by thomas-bc**
In response to https://github.com/Open-MBEE/Comodo/issues/7

## Rationale 
Indeed, the JAR files are not the main reason this repo needs Git LFS. Here's the size of the main folders of the repo:
``` 
 17M	libs
 18M	comodo2
 38M	docs
 76M	test
```

Still, adding a build automation tool such as Gradle and retrieving the dependencies from a repository is a good idea in my opinion. I also plan to add testing through Gradle after we get this reviewed.

## Additional comment
We should probably take the `docs` out of this repository and publish them somewhere else. I'm not sure why all those documents are in here. \
Also, are all those UML files actually being used for testing? I don't know that all the byproducts of the extraction are used by Comodo.

## Future work
- Add testing tasks to Gradle and deprecate the Ant build, to be able to remove the JARs from the `libs/` folder.\
- `docs/` folder should be moved to another repository, and linked from here if they are useful documentation.\
- `test` folder should be cleaned up.

I will volunteer for all these tasks, but I am opening this first PR to get it reviewed ASAP.

### CODE REVIEW

**Comment by thomas-bc** [comment]
@landolfa thoughts? Would you mind reviewing?
Thanks!

File: build.gradle [hunk 1]
```diff
@@ -39,14 +39,19 @@ dependencies {
     
     // Eclipse doesn't manage a Maven repo, so we use this function to get the JAR from direct download
     implementation urlFile('https://download.eclipse.org/modeling/mdt/uml2/updates/5.5/R202211161811/plugins/org.eclipse.uml2.uml_5.5.0.v20221116-1811.jar', 'org.eclipse.uml2.uml_5.5.0')
-
-    // implementation fileTree(dir: 'libs', include: 'org.eclipse.xtend.*.jar')
-    // implementation fileTree(dir: 'libs', include: 'org.eclipse.uml2.*.jar')// {exclude 'META-INF/*.RSA', 'META-INF/*.SF'}
+    implementation urlFile('https://download.eclipse.org/modeling/mdt/uml2/updates/5.5milestones/S202211161811/plugins/org.eclipse.uml2.common_2.5.0.v20221116-1811.jar', 'org.eclipse.uml2.common_2.5.0.jar')
```
Comment (landolfa) [inline build.gradle]: The function urlFile() already adds .jar at the end.

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,65 @@
+plugins {
+    id 'java'
+    id 'application'
+}
+
+version '1.0'
+
+repositories {
+    mavenCentral()
+}
```
Comment (landolfa) [inline build.gradle:10]: May be there is a repo for Eclipse EMF jars.

**Comment by landolfa** [comment]
Hi Thomas,

good work! 
I added a couple of minor comments.

Please go ahead with your planned activities.

Thanks a lot,
Luigi

**Review by landolfa — APPROVED** [review]
Nice work!

File: build.gradle (hunk 2 above)
Comment (thomas-bc) [reply build.gradle:10]: I have looked thouroughly and couldn't find anything. A couple of resources lead me to believe that Eclipse simply does not maintain a Maven repo for their EMF jars. There is this https://repo.eclipse.org/content/, but it only has the older versions just like Maven Central.

File: build.gradle (hunk 1 above)
Comment (thomas-bc) [reply build.gradle]: Fixed

File: build.gradle (hunk 2 above)
Comment (landolfa) [reply build.gradle:10]: ok, anyway we can improve it later if we find a way.

**Comment by landolfa** [comment]
@thomas-bc shall I merge or do you plan more work on this pull reuqest?

**Comment by thomas-bc** [comment]
We are good to merge! I'll make new PRs for the other items

**Pull request merged**